Assets/Scripts/PlayerBase.cs

```diff
@@ -51,4 +54,12 @@ void Update()
     public void Die() {
         Destroy(this);
     }
+
+    private void OnCollisionEnter2D(Collision2D other) {
+        // TODO: more consistent raycast logic needed
+        RaycastHit2D hit = Physics2D.Raycast(transform.position, Vector2.down, 0.6f);
+        if (hit.collider != null && hit.collider.tag == "Ground") {
+            this.grounded = true;
+        }
```

Comment:
not super familiar with using Raycast to handle collision logic. Why does one raycast (which I read as: "is the player about to touch the ground?") instead of checking if the player's RigidBody is touching a ground object?